lib/internal/modules/esm/hooks.js

```diff
@@ -333,6 +336,10 @@ class Hooks {
     };
   }
 
+  resolveSync(_originalSpecifier, _parentURL, _importAssertions) {
+    throw new Error('`resolveSync` is not supported.');
```

Comment:
Throwing `Error` is not allowed—you have to use one of the custom ones. I think perhaps [ERR_METHOD_NOT_IMPLEMENTED](https://nodejs.org/api/errors.html#err_method_not_implemented) is most appropriate.